src/CreateInitializePlugin.ts

```diff
@@ -6,17 +6,25 @@ import flash = require('fastify-flash')
 export function CreateInitializePlugin(passport: Authenticator, options: { userProperty?: string } = {}) {
   return fp(async (fastify) => {
     void fastify.register(flash)
-    fastify.decorateRequest('passport', {
-      getter() {
-        return passport
-      },
-    })
-    fastify.decorateRequest('logIn', logIn)
-    fastify.decorateRequest('login', logIn)
-    fastify.decorateRequest('logOut', logOut)
-    fastify.decorateRequest('logout', logOut)
-    fastify.decorateRequest('isAuthenticated', isAuthenticated)
-    fastify.decorateRequest('isUnauthenticated', isUnauthenticated)
+    fastify.decorateRequest('passport', null)
+    fastify.decorateRequest('logIn', null)
+    fastify.decorateRequest('login', null)
+    fastify.decorateRequest('logOut', null)
+    fastify.decorateRequest('logout', null)
+    fastify.decorateRequest('isAuthenticated', null)
+    fastify.decorateRequest('isUnauthenticated', null)
     fastify.decorateRequest(options.userProperty || 'user', null)
+
+    fastify.addHook('onRequest', (request, _reply, done) => {
+      request.logIn = logIn
+      request.login = logIn
+      request.logOut = logOut
+      request.logout = logOut
+      request.isAuthenticated = isAuthenticated
+      request.isUnauthenticated = isUnauthenticated
+      request.passport = passport
+
+      done()
+    })
```

Comment:
Most of these should not be needed as they are functions. You can also use the getter pattern to avoid the deprecation.

Reply:
Just changed it to use the getter pattern